src/plugins/workspace/server/routes/index.ts

```diff
@@ -91,19 +118,41 @@ export function registerRoutes({
       validate: {
         body: schema.object({
           attributes: workspaceAttributesSchema,
+          permissions: schema.maybe(
+            schema.oneOf([workspacePermission, schema.arrayOf(workspacePermission)])
+          ),
         }),
       },
     },
     router.handleLegacyErrors(async (context, req, res) => {
-      const { attributes } = req.body;
+      const { attributes, permissions: permissionsInRequest } = req.body;
+      const principals = permissionControlClient?.getPrincipalsFromRequest(req);
+      let permissions: WorkspacePermissionItem[] = [];
+      if (permissionsInRequest) {
+        permissions = Array.isArray(permissionsInRequest)
+          ? permissionsInRequest
+          : [permissionsInRequest];
+      }
+
+      // Assign workspace owner to current user
+      if (!!principals?.users?.length) {
+        permissions.push({
+          type: 'user',
+          userId: principals.users[0],
+          modes: [WorkspacePermissionMode.LibraryWrite, WorkspacePermissionMode.Write],
+        });
+      }
 
       const result = await client.create(
         {
           context,
           request: req,
           logger,
         },
-        attributes
+        {
+          ...attributes,
+          ...(isPermissionControlEnabled && permissions.length ? { permissions } : {}),
```

Comment:
what about do the check in the beginning so that this router won't go through the transformer if permission control not enabled?